### ISSUE TITLE
Could you please unseal the Render() of the Panel?

### ISSUE THREAD

**Issue by liwuqingxin** [AvaloniaUI/Avalonia#15935]
### Is your feature request related to a problem? Please describe.

I would like to put separators between some ListBoxItem. Just like : 
`Item | Item | Item`
So I want to extend the StackPanel and render some separators on it. It is the simplest way to do that, I think.

BUT, the Render() of Panel is sealed.

### Describe the solution you'd like

Could you please unseal the Render() of the Panel. I can not come out why it is sealed, in my opinion.

**Comment by AnnDevNet** [comment]
+1 !
I also need this like with wpf.

**Comment by pulsarware** [comment]
+1 me too

**Comment by timunie** [comment]
It is sealed because we may want to change how things work in the future and if it wasn't sealed, that may lead to conflicts and breaking changes. 

Also I suggest to just add the sepator to ContainerItemTheme and if needed hide it for the last item. We have nth child selectors which could be useful here as well.

**Comment by liwuqingxin** [comment]
@timunie Thank your for your reply!

Yes, I do this now. But it feels that putting the separator into the item container template is not lightweight enough. I should at least add a panel and a separator. In addition, I have to think about the layout of the separators in my mind so that they are accurately placed in the center of the two containers, which is troublesome to do. Even if I may need to place similar separators in different types of list controls, I need to modify the control template of each list container.

Unsealing the Panel's Render function will give the Panel more flexible extensibility, although it is optional.

If the user overrides the panel's Render function, then the user needs to be responsible for it. Render will only perform some additional auxiliary drawing and will not have an additional impact on the main function of the Panel-layout, which seems to be no breaking change. In contrast, allowing users to control measurement and layout arrangements seems to be more likely to cause conflicts and destructive changes, but this is inevitable, right?

**Comment by jp2masa** [comment]
> Also I suggest to just add the sepator to ContainerItemTheme and if needed hide it for the last item. We have nth child selectors which could be useful here as well.

Unfortunately, it's not that simple, particularly for selectable items. The separator will visually appear as contained in the selectable/selected item, which does not make sense.

It still doesn't mean that overriding `Render` is required, as the separators can be created as visual children, which is logically the best option, in my opinion. It may also be possible to add one or multiple visual nodes to the compositor tree, but I would not recommend it.

To create the separators as visual children, it is as "simple" as overriding this method appropriately:

https://github.com/AvaloniaUI/Avalonia/blob/82e3613939af5451597d3c19f3a48e0a8ba888f3/src/Avalonia.Controls/Panel.cs#L134

It's not very easy, but it should be maintainable and work well with virtualization if implemented correctly.

**Comment by liwuqingxin** [comment]
> It may also be possible to add one or multiple visual nodes to the compositor tree, but I would not recommend it.
> 
> To create the separators as visual children, it is as "simple" as overriding this method appropriately:
> 
> [Avalonia/src/Avalonia.Controls/Panel.cs](https://github.com/AvaloniaUI/Avalonia/blob/82e3613939af5451597d3c19f3a48e0a8ba888f3/src/Avalonia.Controls/Panel.cs#L134)
> Line 134 in [82e3613](https://github.com/AvaloniaUI/Avalonia/commit/82e3613939af5451597d3c19f3a48e0a8ba888f3)
> protected virtual void ChildrenChanged(object? sender, NotifyCollectionChangedEventArgs e)

That's right. This is a reasonable approach, if we only add separators to the list and do not do other operations such as iteratoring visual children.

Whether the separator is added in the control template or in the Children of the Panel, it is always a Visual in the visual tree. In some complex scenarios, we need to give them more attention.

What I mean is that we can use different ways to achieve this goal in different scenarios. Whether it is overriding the control template, overriding the Panel's ChildrenChanged, or overriding Render, these methods may be suitable for various scenarios.

If the Render method can be unsealed, the Panel will gain greater flexibility, even if this flexibility is not reflected in the scenario of adding separators, in some cases.
